lists-service/src/main/java/au/org/ala/listsapi/service/ConstraintService.java

```diff
@@ -0,0 +1,94 @@
+package au.org.ala.listsapi.service;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+import au.org.ala.listsapi.model.ConstraintListItem;
+import au.org.ala.listsapi.model.ConstraintType;
+import au.org.ala.listsapi.model.InputSpeciesList;
+import au.org.ala.listsapi.model.SpeciesList;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import jakarta.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ConstraintService {
+  private static final Logger logger = LoggerFactory.getLogger(ConstraintService.class);
+
+  @Qualifier("webApplicationContext")
+  @Autowired private ResourceLoader resourceLoader;
+
+  private Map<String, List<ConstraintListItem>> constraints = null;
+
+  private String getJson() {
+    String out = null;
+    Resource resource = resourceLoader.getResource("classpath:/constraints.json");
```

Comment:
This must be a configurable file path for living-atlas deployments. 

Regions should be easy to change or automatic for specific states and countries that are relevant where the app is deployed. I recently found a similar issue addressed in biocache-hubs that makes uses of data in userdetails, https://github.com/AtlasOfLivingAustralia/biocache-hubs/blob/develop/grails-app/services/au/org/ala/biocache/hubs/WebServicesService.groovy#L718 and https://github.com/AtlasOfLivingAustralia/biocache-hubs/blob/develop/grails-app/services/au/org/ala/biocache/hubs/WebServicesService.groovy#L736. This is available in the UI on the page https://biocache.ala.org.au/search#tab_advanceSearch.

Reply:
Dynamic retrieval of countries & state in 22f28e40e617f76eb70e85472c91d0dbbc958af1 - UI implementation TBD, are we replacing the `region` filed with `country` and `state`?